### PR TITLE
Simplified bit operations, cleaned up and added some switch-cases

### PR DESCRIPTION
**Pull request by gbbr**
I hope I haven't broken anything. Let me know if you don't like anything.

### CODE REVIEW

**Comment by gbbr** [comment]
Feel free to close it if you don't like it or if it's terribly wrong. I was bored :)

File: 6502.go [hunk 1]
```diff
@@ -23,13 +23,13 @@ func (p ProcStat) getAsWord() (pstatus int) {
...
-	if pstatus&BIT_4 == 0 { p.b = 1 } else { p.b = 0 }
-	if pstatus&BIT_6 == 0 { p.n = 1 } else { p.n = 0 }
-	if pstatus&BIT_7 == 0 { p.v = 1 } else { p.v = 0 }
+	p.c = ^(pstatus & 1)
+	p.z = ^((pstatus >> 1) & 1)
+	p.i = ^((pstatus >> 2) & 1)
+	p.d = ^((pstatus >> 3) & 1)
+	p.b = ^((pstatus >> 4) & 1)
+	p.n = ^((pstatus >> 6) & 1)
+	p.v = ^((pstatus >> 7) & 1)
```
Comment (gbbr) [inline 6502.go:32]: I think these negations are wrong

File: 6502.go [hunk 2]
```diff
@@ -1420,46 +1359,35 @@ func (cpu *Cpu) rorm(addr int) {
...
 	cpu.pc = ((h << 8) | l) + 1
 }
 
 // substract with carry
 func (cpu *Cpu) sbc(addr int) {
-	data := cpu.mem.read(addr)
-
 	var t int
+	data := cpu.mem.read(addr)
+	negcarry := ^(cpu.p.c & 1)
```
Comment (gbbr) [inline 6502.go:1385]: Same here

File: 6502.go (hunk 1 above)
Comment (jmle) [inline 6502.go:32]: It's wrong because my code is wrong :yum: It should be without the negation

File: 6502.go [hunk 3]
```diff
@@ -186,37 +186,37 @@ func (cpu *Cpu) execute() (resCycles int) {
...
-			} else {
-				resCycles = 3
-			}
-		} else {
+		bcc := cpu.bcc(cpu.rel())
+		switch {
+		case bcc && cpu.pbCrossed:
+			resCycles = 4
+		case bcc:
+			resCycles = 4
```
Comment (jmle) [inline 6502.go:194]: This should be `resCycles = 3`

**Comment by jmle** [comment]
I think the idea of using switches is good. But I think that other changes make the code harder to understand, like for example using the bit shifts instead of the `BIT_*` masks

**Comment by gbbr** [comment]
I agree. I was just playing around. I will remove the bit shift "magic"

**Pull request closed without merging**